test/system/src/test/java/io/pravega/test/system/ControllerFailoverTest.java

```diff
@@ -99,23 +131,38 @@ public static void setup() throws InterruptedException, MarathonException, URISy
         log.debug("pravega host service details: {}", segUris);
     }
 
-    private static URI getTestControllerServiceURI() {
-        Service controllerService = new PravegaControllerService(TEST_CONTROLLER_SERVICE_NAME, null);
-        List<URI> ctlURIs = controllerService.getServiceDetails();
-        return ctlURIs.get(0);
-    }
-
-    private static URI getControllerURI() {
-        Service controllerService = new PravegaControllerService("controller", null);
-        List<URI> ctlURIs = controllerService.getServiceDetails();
-        return ctlURIs.get(0);
-    }
 
+    @Before
+    public void setup() {
+        Service zkService = Utils.isDockerLocalExecEnabled()
+                ? new ZookeeperDockerService("zookeeper")
+                : new ZookeeperService("zookeeper");
+        Assert.assertTrue(zkService.isRunning());
+        List<URI> zkUris = zkService.getServiceDetails();
+        log.info("zookeeper service details: {}", zkUris);
 
-    private static void stopTestControllerService() {
-        log.info("Stopping test controller service");
-        Service controllerService = new PravegaControllerService(TEST_CONTROLLER_SERVICE_NAME, null);
-        controllerService.stop();
+        controllerService1 = Utils.isDockerLocalExecEnabled()
+                ? new PravegaControllerDockerService("controller", zkUris.get(0))
+                : new PravegaControllerService("controller", zkUris.get(0));
+        if (!controllerService1.isRunning()) {
+            controllerService1.start(true);
+        }
+        //controllerService1.scaleService(3, true);
+        List<URI> conUris = controllerService1.getServiceDetails();
+        log.info("conuris {} {}", conUris.get(0), conUris.get(1));
+        log.debug("Pravega Controller service  details: {}", conUris);
+        // Fetch all the RPC endpoints and construct the client URIs.
+        List<String> uris;
+        if (Utils.isDockerLocalExecEnabled()) {
+            uris = conUris.stream().filter(uri -> uri.getPort() ==  Utils.DOCKER_CONTROLLER_PORT).map(URI::getAuthority)
```

Comment:
Why have different ports for Docker v/s marathon?

Reply:
In marathon we had a problem using the default `9090` and hence had to use , `9092`. now , in docker am using the default as the ports are already exposed in docker file.